rholearn/doslearn/train.py

```diff
@@ -77,42 +80,60 @@ def train():
         dtype=ml_options["TRAIN"]["dtype"],
         device=ml_options["TRAIN"]["device"],
     )
-
+    # Should we remove these?
     # Try a model save/load
-    torch.save(model, join(ml_options["ML_DIR"], "model.pt"))
-    torch.load(join(ml_options["ML_DIR"], "model.pt"))
-    os.remove(join(ml_options["ML_DIR"], "model.pt"))
-    io.log(log_path, str(model).replace("\n", f"\n# {utils.timestamp()}"))
-
-    # Compute descriptor
-    io.log(log_path, "Computing SOAP power spectrum descriptors")
-    descriptor = model.compute_descriptor(frames).to(dtype=ml_options["TRAIN"]["dtype"], device=ml_options["TRAIN"]["device"])
-    descriptor = [
-        mts.slice(descriptor, "samples", mts.Labels(["system"], torch.tensor([A]).reshape(-1, 1)))
-        for A in frame_idxs
-    ]
-
+    # torch.save(model, join(ml_options["ML_DIR"], "model.pt"))
+    # torch.load(join(ml_options["ML_DIR"], "model.pt"))
+    # os.remove(join(ml_options["ML_DIR"], "model.pt"))
+    # io.log(log_path, str(model).replace("\n", f"\n# {utils.timestamp()}"))
+
+    # Check if descriptors have been computed
+    descriptor_saved = os.path.exists(join(ml_options["ML_DIR"], "descriptors.pt"))
+    if descriptor_saved:
+        descriptor = torch.load(join(ml_options["ML_DIR"], "descriptors.pt"))
+    else:
+        # Compute descriptor
+        io.log(log_path, "Computing SOAP power spectrum descriptors")
+        descriptor = model.compute_descriptor(frames).to(dtype=ml_options["TRAIN"]["dtype"], device=ml_options["TRAIN"]["device"])
+        descriptor = [
+            mts.slice(descriptor, "samples", mts.Labels(["system"], torch.tensor([A]).reshape(-1, 1)))
+            for A in frame_idxs
+        ]
+        torch.save(descriptor, join(ml_options["ML_DIR"], "descriptors.pt")) # Is this the best way to save it?
```

Comment:
```suggestion
    if (yaml option load descriptors from path)
        descriptors = [load_tensormap(os.path.join(path, frame_idx)) for frame_idx in frame_idxs]
    else
        descriptors = model.compute_descriptor(frames, frame_idxs)
        descriptors = [mts.slice(...) ...]
        if save_descriptors:
            [mts.save(d, "path/to/checkpoint/frame_idx/descriptor.npz") for d in descriptors]
```

Reply:
Removed saving for now, will provide saving support in a future release